gpii/node_modules/flowManager/test/SolutionsRegistryDataSourceTests.js

```diff
@@ -0,0 +1,316 @@
+/*!
+GPII download and save solutions registry from source code repositoyr
+
+Copyright 2020 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion"),
+    kettle = require("kettle"),
+    jqUnit = fluid.registerNamespace("jqUnit"),
+    gpii = fluid.registerNamespace("gpii");
+
+kettle.loadTestingSupport();
+
+fluid.require("%flowManager/src/SolutionsRegistryDataSource.js");
+fluid.registerNamespace("gpii.tests.solutionsRegistry");
+
+gpii.tests.solutionsRegistry = {
+    path: "%gpii-universal/testData/solutions",
+    platformIds: ["win32", "linux", "darwin", "android", "web"],
+    platformId: "darwin",
+    repositoryDarwinSolutions: require("./data/darwin.json5"),
+    requestOptions: { os: "darwin" },
+    requestOptionsMissingOs: {foo: "bar"},
+    win32PlatformId: "win32"
+};
+
+// Check that the local solutions for the given platformId matches the result.
+gpii.tests.solutionsRegistry.checkLocalRegistry = function (msg, platformId, result) {
+    var solutions = fluid.require(gpii.tests.solutionsRegistry.path + "/" + platformId + ".json5");
+    jqUnit.assertDeepEq(msg, solutions, result);
+};
+
+// equalityCheck is either jqUnit.assertDeepEq() or jqUnit.assertDeepNeq().
+gpii.tests.solutionsRegistry.checkNamedRegistry = function (msg, equalityCheck, expectedSolutions, result) {
+    jqUnit.assertValue(msg, result);
+    equalityCheck(msg, expectedSolutions, result);
+};
+
+// Check for failures.
+gpii.tests.solutionsRegistry.checkRejection = function (msg, result) {
+    jqUnit.assertTrue(msg, result.isError);
+    jqUnit.assertEquals(msg, 404, result.statusCode);
+};
+
+// ======== Testing the solutions registry data source used by the CBFM ========
+
+fluid.defaults("gpii.tests.solutionsRegistry.cloud.testCaseHolder", {
+    gradeNames: "fluid.test.testCaseHolder",
+    components: {
+        solutionsRegistryDataSource: {
+            type: "gpii.flowManager.solutionsRegistry.dataSource.cloudBased",
+            options: {
+                path: gpii.tests.solutionsRegistry.path
+            }
+        }
+    },
+    modules: [{
```

Comment:
Please also verify `solutionsRegistryReady` is fired. This comment applies to all test cases.

Reply:
Okay.  I did this by adding a listener to the test SRDS component in the `testCaseHolder` because the event happens long before the test sequence starts up, and listening at that point always misses it.  I hope that's reasonable.